packages/@react-spectrum/table/test/Table.test.js

```diff
@@ -3191,22 +3193,21 @@ export let tableTests = () => {
         });
       });
 
-      it('should support Enter to perform onAction with keyboard', function () {
+      it('should support Enter to perform onAction with keyboard', async function () {
         let onSelectionChange = jest.fn();
         let onAction = jest.fn();
         let tree = renderTable({onSelectionChange, selectionStyle: 'highlight', onAction});
 
-        fireEvent.keyDown(getCell(tree, 'Baz 10'), {key: ' '});
-        fireEvent.keyUp(getCell(tree, 'Baz 10'), {key: ' '});
+        act(() => getCell(tree, 'Baz 10').focus());
+        await user.keyboard(' ');
         checkSelection(onSelectionChange, ['Foo 10']);
-        // screen reader automatically handles this one
-        expect(announce).not.toHaveBeenCalled();
+        expect(announce).toHaveBeenCalledWith('Foo 10 selected.');
```

Comment:
This test was wrong before. The announcement was not fired because the table never had focus.